### PR TITLE
fix: Routing form accepts blank name value

### LINKED ISSUE #9871: Routing form accepts blank as its name
### Issue Summary

To have a name is something mandatory for the routing form, however, it accepts blank as a name.
 
### Steps to Reproduce

1. Install the routing form app
2. Click to add a new routing form
3. Type a single space
4. Click on the Continue button
5. Observe

### Actual Results

- It's accepting blank as a name

### Expected Results

- It should validate that there is no valid value

### Technical details

- Tested on Chrome (114.0.5735.134)

### Evidence

image

image

image

### PR DESCRIPTION
**Pull request by ParbhatKataria1**
This PR makes the "name" value mandatory in the Routing Form.

Fixes #9871 

/claim #9871

Screenshot 2023-07-02 161448

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

1. Install the routing form app
2. Click to add a new routing form
3. Click to add a new routing form
4. Type a single space
5. Click on the Continue button
6. Observe

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

**Comment by PeerRich** [comment]
closing in favor of https://github.com/calcom/cal.com/pull/9882

**Comment by PeerRich** [comment]
just realised "required" is not enough https://github.com/calcom/cal.com/pull/9877#issuecomment-1616704145

File: packages/app-store/routing-forms/components/FormActions.tsx [hunk 1]
```diff
@@ -77,11 +78,17 @@ function NewFormDialog({ appUrl }: { appUrl: string }) {
     },
   });
 
+  const formDataSchema = z.object({
+    name: z.string().trim().min(1, t("title_must_contain_at_least_1_character")),
```
Comment (PeerRich) [inline packages/app-store/routing-forms/components/FormActions.tsx:82]:
```suggestion
    name: z.string().trim().min(1, t("required")),
```

File: apps/web/public/static/locales/en/common.json [hunk 2]
```diff
@@ -1922,5 +1922,6 @@
   "insights_team_filter": "Team: {{teamName}}",
   "insights_user_filter": "User: {{userName}}",
   "insights_subtitle": "View booking insights across your events",
+  "title_must_contain_at_least_1_character":"Title must contain at least 1 character(s)",
```
Comment (PeerRich) [inline apps/web/public/static/locales/en/common.json:1925]:
```suggestion
```

**Review by PeerRich — APPROVED** [review]
i think this is the best

**Review by Udit-takkar — COMMENTED** [review]
@ParbhatKataria fix all the lint errors

**Review by hariombalhara — CHANGES_REQUESTED** [review]
Thanks @ParbhatKataria for the PR. But the solution is incomplete.
During edit of an existing form I can save without providing name.
Also, please check duplication of Routing Form flow as well.

**Pull request closed without merging**